dataquality/loggers/data_logger/structured_classification.py

```diff
@@ -116,58 +116,70 @@ def log(self) -> None:
         Support for batching to come in V1 of structured data project.
 
         We write the dfs to disk in the following locations:
-        /Users/username/.galileo/logs/proj-id/run-id/training/data.hdf5
-        /Users/username/.galileo/logs/proj-id/run-id/training/prob.hdf5
+        /Users/username/.galileo/logs/proj-id/run-id/training/data/data.hdf5
+        /Users/username/.galileo/logs/proj-id/run-id/training/prob/prob.hdf5
+        /Users/username/.galileo/logs/proj-id/run-id/training/features/features.hdf5
 
         NOTE: We don't restrict row or feature counts here for cloud users.
         """
         self.validate_and_prepare_logger()
 
         object_base_path = f"{self.write_output_dir}/{self.split_name_path}"
-        data_df, probs_df = self._get_dfs()
+        df, prob_df, feature_df = self._get_dfs()
 
         # Write DFs to disk
         os.makedirs(f"{object_base_path}/data", exist_ok=True)
-        data_df.export(f"{object_base_path}/data/data.hdf5")
+        df.export(f"{object_base_path}/data/data.hdf5")
         os.makedirs(f"{object_base_path}/prob", exist_ok=True)
-        probs_df.export(f"{object_base_path}/prob/prob.hdf5")
+        prob_df.export(f"{object_base_path}/prob/prob.hdf5")
+        os.makedirs(f"{object_base_path}/features", exist_ok=True)
+        feature_df.export(f"{object_base_path}/features/features.hdf5")
 
-    def _get_dfs(self) -> Tuple[DataFrame, DataFrame]:
-        """Returns data and probs as vaex DataFrames
+    def _get_dfs(self) -> Tuple[DataFrame, DataFrame, DataFrame]:
+        """Returns feature, data and prob as vaex DataFrames
 
-        For tabular data, data DF is just the original input features, with
-        an added "id" column
+        Feature DF is just the input features, with an added "id" column
 
-        Prob DF has:
+        Data DF has:
         - id
-        - prob
+        - pred
         - split
         - data_schema_version
-        - gold (non-inference only)
         - inference_name (inference only)
+
+        Prob DF has:
+        - id
+        - prob
+        - gold (non-inference only)
         """
         assert isinstance(self.X, pd.DataFrame), (
             "X must be a pandas DataFrame. " f"X is currently a {type(self.X)}"
         )
+        assert self.probs is not None, "probs must be set before getting dfs"
+
         ids = self.X.id.values if "id" in self.X.columns else np.arange(len(self.X))
         n_rows = len(self.X)
 
-        df = vaex.from_pandas(self.X)
-        df["id"] = ids
+        feature_df = vaex.from_pandas(self.X)
+        feature_df["id"] = ids
```

Comment:
will `ids` ever be a list? If you're setting this in vaex, itll have to be a numpy array. So you may wanna do 
```suggestion
        feature_df["id"] = np.array(ids)
```
if it could be a list

Reply:
it will always be numpy!